packages/gatsby-transformer-code-examples/gatsby-node.js

```diff
@@ -0,0 +1,80 @@
+const createSchemaCustomization = ({ actions, schema }) => {
+  const { createTypes } = actions;
+
+  const typeDefs = [
+    schema.buildObjectType({
+      name: 'CodeExample',
+      fields: {
+        name: 'String!',
+        extension: 'String!',
+        absolutePath: 'String!',
+        content: 'String!',
+      },
+      interfaces: ['Node'],
+      extensions: {
+        infer: false,
+      },
+    }),
+  ];
+
+  createTypes(typeDefs);
+};
+
+async function onCreateNode({
+  node,
+  actions,
+  loadNodeContent,
+  createNodeId,
+  createContentDigest,
+}) {
+  if (!isValidMediaType(node.internal.mediaType)) return;
+
+  const { createNode, createParentChildLink } = actions;
+
+  function transformObject(obj, id, type) {
+    const codeSampleNode = {
+      ...obj,
+      id,
+      children: [],
+      parent: node.id,
+      internal: {
+        contentDigest: createContentDigest(obj),
+        type,
+      },
+    };
+    createNode(codeSampleNode);
+    createParentChildLink({ parent: node, child: codeSampleNode });
+  }
+
+  const content = await loadNodeContent(node);
+
+  transformObject(
+    {
+      name: node.name,
+      extension: node.extension,
+      absolutePath: node.absolutePath,
+      content,
+    },
+    createNodeId(`${node.id} >>> CodeExample`),
+    'CodeExample'
+  );
+}
+
+function isValidMediaType(mediaType) {
+  // some files have no unique media type, see comments below
+  const validMediaTypes = [
```

Comment:
This variable does not need to be declared inside the function, as it's a constant value. Otherwise the variable is initalized every time the function is called.
Instead, just declared it within the file.

Reply:
What do you mean "within the file"?

Reply:
Declare it as a variable in the `gatsby-node.js`, not within a function.